lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2895,6 +2920,168 @@ static Value BilinearInterpolate(OpBuilder &b,
   return b.create<arith::AddFOp>(loc, left, right);
 }
 
+static Value WeightFunction(OpBuilder &b, Location loc, Value xDistance) {
```

Comment:
We shouldn't need to determine which branch of the piecewise definition to use at runtime. `x_1` and `x1` should only compute `lessEqualOne`, and `x_2` and `x2` should only use `lessThanTwo`. Nothing is going to be outside that range, so we shouldn't need anything else. This might also be the source of lingering numerical inaccuracy on the boundary, since the `xDistance` you are passing here is based on the clamped values, but it is possible that the OOB elements still use the part of the piecewise function that corresponds to the OOB position, while using the input pixel intensity from the nearest in-bounds pixel. 

Also, it will probably be better to have this as a lambda function inside the bicubic interpolate, so we can capture constants instead of rewriting them 16 times.

Reply:
Oh, nevermind, the clamping happens after. Oh well. Should still make the changes mentioned here.